php/php-rr-sdk.md

```diff
@@ -0,0 +1,692 @@
+of codebase and annotations. No common interface required.
+
+```php
+class FileController 
+{
+    /**
+     * @Workflow\Activity("name" = "file.download") 
+     */         
+    public function processFile(string $file): string 
+    {
+        $result = $this->doProcessing($file);
+
+        return $result;
+    }
+
+    // ...
+}
+```
+
+> PHP 8 annotations or alternative registration methods possible as well.
+
+### Logging
+It is possible to provide access to log stream from the `ActivityContextInterface` which makes possible to use PSR-3 compatible
+loggers.
+
+```php
+$logger = new ActivityLogger($ctx);
+$logger->debug('Hello!');
+
+// or directly
+$ctx->log('debug', 'message', ['context']);
+```
+
+### Payloads
+The payloads and responses can be serialized and deserialized to the target format using simple type reflection. 
+Such an approach makes possible to use PHP objects as data carrying medium:
+
+```php
+class MyObj 
+{
+    public string $something;
+}
+
+function doSomething(ActivityContextInterface $ctx, MyObj $input): Response
+{
+  // ...
+    return new Response('something');
+}
+```
+
+It is mandatory to use proper type declarations on input and return types of activity. While PHP 7.4 is not widely adopted,
+it is recommended to use strict property typing for payload properties as well when possible or highlight the type via
+mapper specific annotation.
+
+### Process Isolation
+Activity handler will stay in memory permanently, only the execution context will change. Such an approach will ellimitate 
+the framework initialization overhead and reduce memory consumption. 
+
+The system must limit to only one concurrent execution per worker, which makes possible to use all existing PHP SPL and vendor
+libraries without changes. 
+
+## Workflows
+Unlike activities, the workflow execution must be based on a physical memory object. In order to ensure proper density
+and effectiveness of the system, multiple workflows will locate inside singe PHP process. Each workflow will operate
+as one or multiple [Generators](https://www.php.net/manual/en/class.generator.php). The execution can be controlled using
+system [keyword](https://www.php.net/manual/en/language.generators.syntax.php) `yield`.
+
+Yield keyword should be used to request the command execution by Temporal system. For the sake of simplify each command
+must be presented as typed object, though, end user API can encapsulate object construction using helper functions. 
+
+Each yield command will return Promise/Future object which can later be referenced in a code. Simple workflow execution might
+look as following:
+
+```php
+use Temporal\Workflow;
+
+class UploadWorkflow
+{
+    public function run(Workflow\ContextInterface $ctx, string $input): string
+    {
+        $promise = yield new Workflow\ExecuteActivy(
+            'activityName',
+            $input, 
+            100 // startToFinishTimeout, etc. 
+        );        
```

Comment:
I believe any activity invocation should get ctx as a parameter.

Reply:
See above, the $ctx always passed inexplicably due to the model of how cooperative multitasking works in PHP. So we can make this DSL smaller and easier to understand.